src/HJSON/HJSONParser.php

```diff
@@ -331,7 +333,7 @@ private function keyname()
 
         while (true) {
             if ($this->ch === ':') {
-                if (!$name) $this->error("Found ':' but no key name (for an empty key name use quotes)");
+                if (!$name && $name != 0) $this->error("Found ':' but no key name (for an empty key name use quotes)");
```

Comment:
Could this condition just be `if ($name === '')`?

Reply:
Yes. It will make this condition clear and simple to know the logic :+1: .